src/app.js

```diff
@@ -69,6 +70,16 @@ app.use(middleware.rejectParentTraversals);
  */
 app.use((req, res, next) => {
   req.context = {};
+
+  // Set the app's origin centrally so other handlers can use it
+  //
+  // We can trust the HTTP Host header (req.hostname) because we're always
+  // running behind name-based virtual hosting on Heroku.  A forged Host header
+  // will be rejected by Heroku and never make it to us.
+  req.context.origin = PRODUCTION
+    ? `${req.protocol}://${req.hostname}`
+    : `${req.protocol}://${req.hostname}:${req.app.get("port")}`;
+
   next();
```

Comment:
Is there a reason for `next()` vs. `return next()` here?

Reply:
Nope. It's equivalent in this context as the final statement, so just style. I would probably write `return next()` most times now, but the existing code omitted the `return`.